src/rholang-server.ts

```diff
@@ -225,18 +234,24 @@ export class RholangServer {
       '-p', `${port0}`, '--http-port', `${port3}`, '--kademlia-port', `${port4}`,
     ]
 
-    // Start RNode (standalone) process used by the server
-    let vm: ChildProcess
-    if (this.useDocker()) {
+    const dockerCmd = runOptions => {
       const volume = `${workingFolder}:/vscode`
-      vm = spawn('docker', [
+      let cmd = [
         // Docker run
         'run', '-i', '--rm', '--name', rhoTmpName, '-v', volume,
                '-p', `${port0}:${port0}`, '-p', `${port1}:${port1}`, '-p', `${port2}:${port2}`,
                '-p', `${port3}:${port3}`, '-p', `${port4}:${port4}`,
-        // RNode run
-        this._settings.rnodeDockerImage, ...runCmd('/vscode/.rnode'),
-      ])
+       ]
+      runOptions && runOptions.length != 0 && cmd.push(`${runOptions}`)
```

Comment:
This is tricky part. `spawn` accepts arguments separated in a list and not in one string. But just to split by space can also be problematic if space is part of the value.

I wrote helper function to parse arguments by `-` to an object.
https://github.com/tgrospic/rnode-grpc-js/blob/a6ae689/src/cli/args.js#L30

This is how it's used. For now object can be transformed to a list and just appended. Later we can add merging of objects.
https://github.com/tgrospic/rnode-grpc-js/blob/a6ae6894/src/cli/index.js#L60

Reply:
I'm not sure we can use your function. I have an option like --network=host but your function strips the dashes. I need an array like ['--network', 'host'] not ['network','host']. Am I understanding that you want a modified version of your version to return lists of string arrays with the dashes intact?

Reply:
Yes, you are right. With equal sign will not work. And also multiple parameters like `-p 123 -p 456` are generated as a list for this key `{ p: [123, 456] }`.
Maybe there already exists some nice lib for what we need here?

Reply:
elsewhere in vs-code config, JSON is used to represent structured args.